src/mlpack/methods/neighbor_search/knn_main.cpp

```diff
@@ -78,6 +82,9 @@ PARAM_FLAG("single_mode", "If true, single-tree search is used (as opposed to "
     "dual-tree search).", "S");
 PARAM_DOUBLE_IN("epsilon", "If specified, will do approximate nearest neighbor "
     "search with given relative error.", "e", 0);
+PARAM_STRING_IN("effective_error", "If specified, will compare the results "
+    "against the provided distances file, and will print the average relative "
+    "error.", "E", "");
```

Comment:
I think we should name this similarly to the parameter to do this for LSH:

```
// For testing recall.
PARAM_STRING_IN("true_neighbors_file", "File of true neighbors to compute "
    "recall with (the recall is printed when -v is specified).", "t", "");
```

In this case you're calculating and printing a slightly different comment, so you'll have to update the comments a bit.  I think you'll need `true_distances_file` instead, and the short option will need to be `-T` since `-t` is already taken.